torch/distributed/device_mesh.py

```diff
@@ -430,21 +430,43 @@ def get_group(
                 return dim_groups
 
         @staticmethod
-        def from_group(group: ProcessGroup, device_type: str) -> "DeviceMesh":
+        def from_group(
+            group: ProcessGroup,
+            device_type: str,
+            mesh_shape: Optional[Tuple[int, ...]] = None,
+            *,
+            mesh_dim_names: Optional[Tuple[str, ...]] = None,
+        ) -> "DeviceMesh":
             """
             Contstructs a :class:`DeviceMesh` with ``device_type`` from an
             existing :class:`ProcessGroup`.
 
-            The constructed device mesh is assumed to be 1D.
+            The constructed device mesh is assumed to be 1D if ``mesh_shape``
+            is not specified. Otherwise, ``mesh_shape`` is used.
             """
-            # Manually define `_dim_group_infos` instead of relying on the
-            # normal logic since we already have the PG
             group_ranks = get_process_group_ranks(group)
-            mesh = DeviceMesh(device_type, group_ranks, _init_backend=False)
-            mesh._dim_group_infos = [
-                (_get_group_tag(group), group_ranks, group.group_name)
-            ]
-            return mesh
+            if mesh_shape is not None:
+                if math.prod(mesh_shape) != len(group_ranks):
+                    raise ValueError(
+                        f"Mesh shape {mesh_shape} is invalid for group with ranks {group_ranks}"
+                    )
+                mesh: Union[torch.Tensor, List[int]] = torch.tensor(
```

Comment:
nit: we always make mesh to be constructed with `torch.tensor` so that type annotation aligns, and only when there's `mesh_shape`, we `view`

Reply:
fixed these locally but did not repush since the approach itself seems flawed now